### PR TITLE
Create a new version of consumption, with preview budget merge with usage detail and reservation

### PR DESCRIPTION
**Pull request by bgsky**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by marstr** [comment]
We made some updates in December, and moved back to working out of `master`. `current` is just a duplicate of `master` to allow for an easier transition. You can read more about all of the changes we made here: [December 2017 Refactoring](https://github.com/Azure/azure-rest-api-specs/wiki/December-2017-Refactoring)

**Comment by bgsky** [comment]
I saw you switched base branch to master. Other than that, let me know what to do with branch on my side and if you have any comments on this pull request.

**Comment by marstr** [comment]
I was just swooping in to make sure the PR was targeting the correct branch. I'll hand this back-off to the assignee, @alvadb.

**Comment by sandeepnl** [comment]
@alvadb @salameer  this PR is been pending for more than a day here. Can someone please review this soon?

**Review by jhendrixMSFT — CHANGES_REQUESTED** [review]
The model validation failures need to be fixed, please see the travis [log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/330537508).

**Comment by jhendrixMSFT** [comment]
@bgsky has this been reviewed and signed off by ARM?

**Comment by bgsky** [comment]
@jhendrixMSFT This is merge of previous preview version with stable version. We don't have a ARM review for this pull request yet.

**Comment by sameralameer** [comment]
Re-assigning to @jhendrixMSFT

**Comment by jhendrixMSFT** [comment]
Can you please provide a link to the preview version from which this swagger originates?  Can you also please provide a reference to the PR for the preview version where ARM team signed off?

**Comment by bgsky** [comment]
Please refer to the following link for the preview version:

https://github.com/Azure/azure-rest-api-specs/pull/2144

**Comment by sandeepnl** [comment]
@jhendrixMSFT @salameer 

To be clear on the ARM review, this PR is publishing a new version of the api with no changes than what is already GA’ed in our ‘2017-11-30’ api version. We had got it reviewed with ARM team then and nothing has changed on it since then. Should we still get another approval from ARM for this too?

Essentially we are combining the operations available in our swagger versions ‘2017-11-30’ and ‘2017-12-30-preview’ into this.

Please let us know

**Comment by jhendrixMSFT** [comment]
@sandeepnl If the ARM team has signed off on the changes for preview then as long as this content is 100% identical there is no need for another review with ARM.  It is good to have all the links to previous PRs with ARM's signoff however in https://github.com/Azure/azure-rest-api-specs/pull/2144 it would appear that the ARM sign-off happened offline as I see no record of it in the PR.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-01-31/consumption.json [hunk 1]
```diff
@@ -853,7 +853,7 @@
       "type": "object",
       "allOf": [
         {
-          "$ref": "#/definitions/ProxyResource"
+          "$ref": "#/definitions/Resource"
```
Comment (jhendrixMSFT) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-01-31/consumption.json]: Was this change intentional?

**Comment by jhendrixMSFT** [comment]
I merged the two swaggers by hand and diff'ed them with what's presented here and there are a few discrepancies.
- Path "/providers/Microsoft.Consumption/operations" is missing in this new version
- Type ProxyResource was renamed to Resource
- The definition for Resource has changed, specifically the "eTag" field was replaced with the "tags" field, why is that?

**Comment by bgsky** [comment]
Thanks for merging and comparing. Will fix errors and send another iteration

**Comment by sandeepnl** [comment]
@bgsky We should have the operations path in the new swagger as well. Please add them and let us talk internally about why etags was removed.
@jhendrixMSFT Thanks for comparing and letting us know.

**Comment by sandeepnl** [comment]
@jhendrixMSFT I see that the operations path is at this location - [Operations](https://github.com/Azure/azure-rest-api-specs/pull/2289/files#diff-823b312a7b7d9f3c3c976f374ba46fbcR446)
Let us know if its not correct

**Comment by bgsky** [comment]
@jhendrixMSFT ProxyResource was used in preview version with etag, while tags as part of stable version. Will fix it by keeping both as original version

**Comment by jhendrixMSFT** [comment]
Good catch on the operations path.  Regarding "eTag" and "tags" does this mean that both properties will be in "Resource"?

**Comment by bgsky** [comment]
"tags" is in "Resource" for one API of usage detail in stable version, and "eTag" in "ProxyResource" for another API of budget in preview. Will keep it the same way

**Comment by jhendrixMSFT** [comment]
@bgsky There is one model validation [failure](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/332465696), can you please take a look?

**Comment by jhendrixMSFT** [comment]
In case you didn't know, you can run these validations locally, see the [docs](https://github.com/Azure/adx-documentation-pr/blob/master/tools/tools.md).

**Comment by bgsky** [comment]
@jhendrixMSFT Thanks, looking into it

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-01-31/examples/UsageDetailsExpand.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
+{
+  "parameters": {
+    "api-version": "2018-01-31",
+    "scope": "subscriptions/subid/providers/Microsoft.Billing/billingPeriods/201702",
+    "$expand": "meterDetails,additionalProperties",
+    "$filter": "usageEnd le 2017-02-14T00:00:00Z",
+    "$top": "1"
```
Comment (sandeepnl) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-01-31/examples/UsageDetailsExpand.json]: @bgsky Change this to integer instead of a string

Comment (bgsky) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-01-31/examples/UsageDetailsExpand.json]: Got it

**Comment by bgsky** [comment]
@jhendrixMSFT Please lets know if model validation works.

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2017-11-30/examples/UsageDetailsExpand.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
     "scope": "subscriptions/subid/providers/Microsoft.Billing/billingPeriods/201702",
     "$expand": "meterDetails,additionalProperties",
     "$filter": "usageEnd le 2017-02-14T00:00:00Z",
-    "$top": "1"
+    "$top": 1
```
Comment (sandeepnl) [inline specification/consumption/resource-manager/Microsoft.Consumption/stable/2017-11-30/examples/UsageDetailsExpand.json]: Dont change this version of the swagger/examples. You need to change it in 2018-01-31 version

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-01-31/examples/UsageDetailsExpand.json (hunk 2 above)
Comment (sandeepnl) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2018-01-31/examples/UsageDetailsExpand.json]: Please change it here

File: specification/consumption/resource-manager/Microsoft.Consumption/stable/2017-11-30/examples/UsageDetailsExpand.json (hunk 3 above)
Comment (bgsky) [reply specification/consumption/resource-manager/Microsoft.Consumption/stable/2017-11-30/examples/UsageDetailsExpand.json]: Thanks for catch. This change is reverted, and new one on latest version is added

**Comment by jhendrixMSFT** [comment]
@dsgouda do you know why this model validation is failing?  It says "expected type integer but found type string" however the type in the example is type integer.  Is this a bug or am I missing something?

**Comment by dsgouda** [comment]
@jhendrixMSFT are these introduced in the current PR? I can't see anything in `consumption.json`.
I'd say @amarzavery or @mcardosos  might have a better idea about model validator.

**Comment by jhendrixMSFT** [comment]
Yep this is current, here's the [travis log](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/332547499).  The operation ID in question is UsageDetails_List, example UsageDetailsExpand.

**Comment by dsgouda** [comment]
😆 I see what's wrong, OpenAPI specs do not recognize `integer`, use `number` instead

**Comment by bgsky** [comment]
Travis CI build does not start automatically after the latest iteration. Is there an issue somewhere?

**Comment by dsgouda** [comment]
Taking a look.

**Comment by jhendrixMSFT** [comment]
@dsgouda That link is for OpenAPI 3.0 which AutoRest doesn't support (yet).  Here's the link to the [2.0 spec](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).  Integer is a valid type and is used in numerous swaggers.  I think the bug is elsewhere, likely the model validator.  Who owns that?

**Comment by jhendrixMSFT** [comment]
I would also add that changing it to number still fails model validation in the same way but with the number type, "Expected type number but found type string".

**Comment by dsgouda** [comment]
You're right @jhendrixMSFT my bad. 
Yes, looks like a model validator bug after all. @mcardosos and @veronicagg own the model validator AFAIK.

**Comment by jhendrixMSFT** [comment]
@bgsky can you please revert your latest commit so they type is changed back to number?  Sorry for the confusion.
@mcardosos is this a known issue with the model validator?

**Comment by bgsky** [comment]
@jhendrixMSFT  will revert it

**Comment by jhendrixMSFT** [comment]
Yes the type should be integer (number is used for floating point/decimal values).

**Comment by jhendrixMSFT** [comment]
I think the model validation failure is a bug, I've opened https://github.com/Azure/oav/issues/197 to track it.

**Review by jhendrixMSFT — APPROVED** [review]

**Comment by lmazuel** [comment]
@jhendrixMSFT This contains a global "budgetName" that should be a method declaration.
https://github.com/Azure/azure-openapi-validator/issues/84

**Comment by jhendrixMSFT** [comment]
@lmazuel All I see is ```budgetNameParameter``` which is in the path.  Am I missing something?
